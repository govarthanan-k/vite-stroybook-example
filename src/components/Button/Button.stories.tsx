import type { Meta, StoryObj } from "@storybook/react-vite";
import { Button } from "./Button";

export default {
  title: "Example/Button",
  component: Button,
} as Meta<typeof Button>;

export const Primary: StoryObj<typeof Button> = {
  args: {
    label: "Click me",
  },
};
