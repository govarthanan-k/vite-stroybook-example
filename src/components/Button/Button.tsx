export interface ButtonProps {
  label: string;
  onClick?: () => void;
}

export function Button({ label, onClick }: ButtonProps) {
  return (
    <button
      data-testid="button"
      style={{
        backgroundColor: "blue",
        color: "#fff",
        border: "none",
        padding: "10px 16px",
        borderRadius: "4px",
        cursor: "pointer",
        fontSize: "16px",
      }}
      onClick={onClick}
    >
      {label}
    </button>
  );
}
