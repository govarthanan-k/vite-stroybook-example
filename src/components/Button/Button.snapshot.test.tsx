import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import { composeStories } from "@storybook/react-vite";
import * as stories from "./Button.stories";

const { Primary } = composeStories(stories);

test("Primary Button Snapshot", () => {
  const { container } = render(<Primary />);
  expect(container.firstChild).toMatchSnapshot();
});
