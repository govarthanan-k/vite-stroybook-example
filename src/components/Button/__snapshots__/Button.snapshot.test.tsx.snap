// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Primary Button Snapshot 1`] = `
<button
  style="background-color: blue; color: rgb(255, 255, 255); padding: 10px 16px; border-radius: 4px; cursor: pointer; font-size: 16px;"
>
  Click me
</button>
`;
